cmds/core/uptime/uptime_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"testing"
+)
+
+type UptimeOutput struct {
+	uptime string
+	err    error
+}
+
+type LoadAvgOutput struct {
+	loadavg string
+	err     error
+}
+
+func TestUptimeGoodData(t *testing.T) {
+	want := UptimeOutput{"up 14 days, 5 hours, 35 minutes ,49 seconds", nil}
+	uptime, err := uptime("1229749 1422244")
+	got := UptimeOutput{uptime, err}
+	if got.uptime != want.uptime {
+		t.Errorf("the uptime got is not correct:want %v , got %v", want.uptime, got.uptime)
+	}
+	if got.err != nil {
+		t.Errorf("expected no error from a good proc/uptime file but got an error:%v", got.err)
+	}
+}
+
+func TestUptimeBadData(t *testing.T) {
+	uptime, err := uptime("string")
+	got := UptimeOutput{uptime, err}
+
+	if got.err == nil {
+		t.Error("expected an invalid contents error but the test did not give an error")
+	}
+}
+
+func TestUptimeEmptyData(t *testing.T) {
+	uptime, err := uptime(" ")
+	got := UptimeOutput{uptime, err}
+
+	if got.err == nil {
```

Comment:
Likewise. Check that you're actually getting the correct error back.

Reply:
Done